src/lightning/pytorch/strategies/xla_fsdp.py

```diff
@@ -0,0 +1,573 @@
+                raise NotImplementedError("XLAFSDP only supports a single model instance with 'model' as the key.")
+            full_ckpt = torch.load(path)
+            model.load_state_dict(full_ckpt.pop("model"), strict=strict)
+            return full_ckpt
+
+        raise ValueError(f"Unknown state_dict_type: {self._state_dict_type}")
+
+    def all_gather(self, tensor: Tensor, group: Optional[Any] = None, sync_grads: bool = False) -> Tensor:
+        """Function to gather a tensor from several distributed processes.
+
+        Args:
+            tensor: tensor to all-gather.
+            group: unused.
+            sync_grads: flag that allows users to synchronize gradients for the all-gather operation.
+        Return:
+            A tensor of shape (world_size, ...)
+
+        """
+        if not self._launched:
+            return tensor
+        if not isinstance(tensor, Tensor):
+            raise NotImplementedError(
+                f"`{type(self).__name__}.all_gather` is only implemented for tensors. Given {tensor}"
+            )
+        if tensor.dim() == 0:
+            tensor = tensor.unsqueeze(0)
+        original_device = tensor.device
+        tensor = tensor.to(self.root_device)
+
+        import torch_xla.core.functions as xf
+        import torch_xla.core.xla_model as xm
+
+        tensor = xf.all_gather(tensor) if sync_grads else xm.all_gather(tensor)
+        tensor = tensor.to(original_device)
+        return tensor
+
+    @classmethod
+    def register_strategies(cls, strategy_registry: _StrategyRegistry) -> None:
+        strategy_registry.register(
+            "xla_fsdp",
+            cls,
+            description=cls.__class__.__name__,
+        )
+
+    def _parse_fsdp_kwargs(self) -> Dict:
+        # this needs to be delayed because `self.precision` isn't available at init
+        kwargs = self._fsdp_kwargs.copy()
+        precision_plugin = self.precision_plugin
+        if isinstance(precision_plugin, XLAPrecisionPlugin):
+            # the `compute_dtype` will be passed to the `auto_wrapper_callable` automatically, so we don't need to pass
+            # it when creating it
+            kwargs.setdefault("compute_dtype", precision_plugin._desired_dtype)
+        kwargs = _auto_wrap_policy_kwargs(self._auto_wrap_policy, kwargs)
+        return _activation_checkpointing_kwargs(self._activation_checkpointing_policy, kwargs)
+
+    def _pod_progress_bar_force_stdout(self) -> None:
+        # Why is it required? The way `pytorch_xla.distributed` streams logs
+        # from different vms to the main worker doesn't work well with tqdm
+        # Ref: https://github.com/pytorch/xla/blob/master/torch_xla/distributed/xla_dist.py#L140
+        # The print statement seems to force tqdm to flush stdout.
+        import torch_xla.core.xla_env_vars as xenv
+        from torch_xla.utils.utils import getenv_as
+
+        if self.global_rank == 0 and getenv_as(xenv.TPUVM_MODE, int, 0) == 1:
+            print()
```

Comment:
I don't think this is required with PjRT. Opened https://github.com/Lightning-AI/lightning/pull/18765

Reply:
cc @Liyang90 @jackCaoG